apps/site/assets/ts/stop/components/StopPageRedesign.tsx

```diff
@@ -6,19 +6,25 @@ import { useRoutesByStop } from "../../hooks/useRoute";
 import StopPageHeaderRedesign from "./StopPageHeaderRedesign";
 import Loading from "../../components/Loading";
 import StopPageDepartures from "./StopPageDepartures";
+import { isASilverLineRoute } from "../../models/route";
 
 const StopPageRedesign = ({
   stopId
 }: {
   stopId: string;
 }): ReactElement<HTMLElement> => {
   const stop = useStop(stopId);
-  const routes = useRoutesByStop(stopId);
-
+  const routesWithPolylines = useRoutesByStop(stopId);
   // Return loading indicator while waiting on data fetch
-  if (!stop || !routes) {
+  if (!stop || !routesWithPolylines) {
     return <Loading />;
   }
+  const routes = routesWithPolylines.map(([route]) => route);
+  const polylines = routesWithPolylines
+    .filter(
+      ([route]) => [0, 1, 2].includes(route.type) || isASilverLineRoute(route)
```

Comment:
question: why not do this filtering on the backend?

Reply:
Yeah, why not! I've made this change.